AP-protocol.md

```diff
@@ -157,141 +166,87 @@ Message is subject to fragmentation: see the dedicated section.
 Message (and Message only) can be dropped by drones.
 
 ```rust
-struct Message{
-	message_header: MessageHeader,
-	/// Shows the type of the message and contains the message.
-	message_content: MessageContent,
-	/// Used and modified by drones to route the packet.
-	source_routing_header: SourceRoutingHeader
+// Server is multype
+#[derive(Debug)]
+pub struct ServerType{ // 1 or more must be true
+	is_chat_server: bool,
+	is_text_server: bool, // must be true if media is true
+	is_media_server: bool
 }
 
-// The serialized and possibly fragmented message sent by
-// either the client or server identified by source_id.
-struct MessageHeader {
-	/// ID of client or server
-	source_id: Option<u64>,
+#[derive(Debug)]
+pub struct Message{
+	message_data: MessageData,
+	routing_header: SourceRoutingHeader
 }
 
-enum MessageContent{
-	ChatMessage(ChatMessage), // chat == communication server
-	TextMessage(TextMessage), // text == content server
-	MediaMessage(MediaMessage)// media == content server
-}
-
-enum ChatMessage{
-	ChatRequest(ChatRequest),
-	ChatResponse(ChatResponse)
-}
-
-enum ChatRequest{ //(chat == communication server)
-	ClientList, // => C -> S : client_list?
-	MessageFor { // => C -> S : message_for?(client_id, message)
-		// note: message_size omitted!
-		client: u64,
-		message: String,
-	},
-}
-
-enum ChatResponse{
-	ClientList(u64, Vec<u64>), //=> S -> C : client_list!(list_length, list_of_client_ids)
-	MessageFrom{ // => S -> C : message_from!(client_id, message)
-		// note: message_size omitted!
-		client: u64,
-		message: String
-	},
-	ErrWrongClient // => S -> C : error_wrong_client_id!
-}
-
-enum TextMessage{// (text == media server with text)
-	TextRequest(TextRequest),
-	TextResponse(TextResponse)
-}
-
-enum TextRequest{
-	ServerType, //C -> S : server_type?
-	FilesList, //C -> S : files_list?
-	File(file_id),// C -> S : file?(file_id)    note: additional params omitted!
-}
-
-enum TextResponse{
-	ServerType(ServerKind), //S -> C : server_type!(type)
-	FilesListResponse { //S -> C : files_list!(list_length, list_of_file_ids)
-		list_length: char,
-		list_of_file_ids: [u64,20],
-	},
-	ErrorNoFiles, // S -> C : error_no_files!
-	File{ //S -> C : file!(file_size, file)
-		file_size: u64,
-		file: String,
-	},
-	ErrorFileNotFound, //S -> C : error_file_not_found!
-}
-
-enum MediaMessage{
-	MediaRequest(MediaRequest),
-	MediaResponse(MediaResponse)
-}
-
-struct MediaRequest{
-	Media{  //C -> S : media?(media_id, +media type+ )
-		media_id: u64
-		///media: media_kind
-	}
+#[derive(Debug)]
+pub struct MessageData { // Only part fragmentized
+	source_id: NodeId,
```

Comment:
```suggestion
```
The `source_id` is already present into the `SourceRoutingHeader`.

Reply:
Fair. I'll remove it now. i still think it is still may be better to keep it here to isolate different layer. But i'll open a PR about it.